test/tictactoe/game_menu_test.clj

```diff
@@ -0,0 +1,54 @@
+(ns tictactoe.game-menu-test
+  (:require [clojure.test :refer :all]
+            [tictactoe.game-menu :refer :all]
+            [tictactoe.game-messages :as messages]
+            [tictactoe.user-interface :as ui]))
+
+(def game-types
+  {:1 "Human vs. Human" :2 "Human vs. Easy Computer"})
+
+(def board-sizes
+  {:1 "3x3" :2 "4x4"})
+
+(defn make-input [coll]
+  (apply str (interleave coll (repeat "\n"))))
+           
+(deftest show-welcome-message
+  (testing "Welcome message is shown"
+    (is (= "Welcome to Tic Tac Toe!\n\n" (with-out-str (welcome-message))))))
+
+(deftest get-game-type-from-user 
+  (testing "Returns input as keyword" 
+    (with-out-str "test" (ui/print-message (messages/game-type))
+      (is (= :1 (with-in-str "1" (pick game-types (messages/game-type))))))))
+
+(deftest get-board-size-from-user
+  (testing "Returns input as keyword when valid choice"
+    (with-out-str "test" (ui/print-message (messages/board-type))
+      (is (= :1 (with-in-str "1" (pick board-sizes (messages/board-type))))))))
+
+(deftest get-4x4-board-size-from-user
+  (testing "Get second board option from user; returns choice as keyword"
+    (with-out-str "test" (ui/print-message (messages/board-type))
+      (is (= :2 (with-in-str "2" (pick board-sizes (messages/board-type))))))))
+
+(deftest get-player-one-marker
+  (testing "Get first players marker and return it"
+    (with-out-str "test" (ui/print-message (messages/player-one-marker))
+      (is (= "X" (with-in-str "X" (pick-marker)))))))
+
+(deftest get-player-two-marker
+  (testing "Get second players marker and return it"
+	 (with-out-str "test" (ui/print-message (messages/player-two-marker))
```

Comment:
Some strange formatting here. and below. If you're in vim, a lot of times I'll just select the whole file and press `=` to reformat.

Reply:
`gg` go to top
`V` to select the first line
`G` to select all lines
`=` to reindent